.github/workflows/ci.yml

```diff
@@ -898,14 +899,81 @@ jobs:
           cd semver
           rm output
           rm -f unexpectedly_did_not_fail
-
+      
       - name: Save rustdoc
         uses: actions/cache/save@v3
         if: steps.cache.outputs.cache-hit != 'true'
         with:
           key: ${{ runner.os }}-${{ hashFiles('.rustc-version', 'semver/**/Cargo.lock') }}-${{ github.job }}-rustdoc
           path: subject/target/semver-checks/cache
 
+  run-on-ref-slice-fork-windows:
+    # Same as above, but run on a Windows machine and from a compiled binary after
+    # removing .cargo/registry/index. This broke in a workflow of the GitHub Action
+    # using a precompiled binary after a fresh Rust installation, when the
+    # registry index does not exist yet.
+    name: Run cargo-semver-checks on ref-slice fork (Windows)
+    runs-on: windows-latest
+    steps:
+      - name: Checkout cargo-semver-checks
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: true
+
+      - name: Checkout ref-slice fork semver break
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+          repository: 'mgr0dzicki/cargo-semver-action-ref-slice'
+          ref: 'major_change'
+          path: 'subject'
+
+      - name: Install rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
+
+      - uses: Swatinem/rust-cache@v2
+        with:
+          key: ref-slice-fork-windows
+
+      - name: Build cargo-semver-checks
+        run: cargo build
+
+      - name: Remove the cargo registry index
+        run: Remove-Item -Recurse -Force C:\Users\runneradmin\.cargo\registry\index
```

Comment:
Out of curiosity, why do we need to forcibly remove this directory to cause this problem here? Is there a situation where the directory doesn't already exist even though the user has a working Rust environment installed on their machine?

I'm just trying to figure out whether this is a bug in the GitHub Action, in `cargo-semver-checks` itself, in one of our dependency libraries, or in `cargo`.

Reply:
The directory is not created during Rust installation, but only after building some crate. Hence if we have a fresh Rust installation and use the downloaded binary of `cargo-semver-checks`, the `index` directory does not exist yet. And this is exactly the case inside GitHub Action.

Here the tool needs to be built first, as we don't want to use the latest release, but the source code which is currently in the repository. But then `index` directory is created during the built, so I remove it afterwards to simulate the environment before building any crate. 

The issue itself is that creating a subdirectory of `index` (for example `index\github.com-1ecc6299db9ec823`) fails, because `index` does not exist. It seems that for some reason it is automatically created along with the desired subdirectory on Ubuntu and MacOS, but not on Windows.

Reply:
Very interesting, nice catch!

Reply:
Opened https://github.com/frewsxcv/rust-crates-index/issues/97 for this. We should also merge this PR, just to be extra safe.